### ISSUE TITLE
Remove redundant quota check

### ISSUE THREAD

**Issue by sedkis** [TykTechnologies/tyk#2777]
related to #2780 
**Branch/Environment/Version**
- Branch/Version: All
- Environment: All

**Describe the bug**
When retrieving a Key, Tyk will attempt to check for a Quota pool, despite it being disabled.  I've disabled the Quota by giving it a value of "-1"

**Reproduction steps**
1.  Create a Key, with Quota Max value "-1" (disabled)
2.  Get key details using the API 
`/api/apis/927605639bc54ee346956bba1742a1ce/keys/5d431f8d06ac9e183bf49f9c1d5a1e3e458d4349bf048b3d603cc240`

**Actual behavior**
Check logs, you will see this:
```
[Jan  3 16:23:49]  INFO api: Can't retrieve key quota error=key not found key=****c240 status=ok
[Jan  3 16:23:49]  INFO api: Can't retrieve api limit quota apiID=927605639bc54ee346956bba1742a1ce error=key not found key=****c240
[Jan  3 16:23:49]  INFO api: Retrieved key detail. key=****c240 status=ok
[Jan  3 16:23:55]  INFO api: Can't retrieve key quota error=key not found key=****c240 status=ok
[Jan  3 16:23:55]  INFO api: Can't retrieve api limit quota apiID=927605639bc54ee346956bba1742a1ce error=key not found key=****c240
[Jan  3 16:23:55]  INFO api: Retrieved key detail. key=****c240 status=ok
```

**Expected behavior**
I don't want Tyk to check for Quota if I have it disabled

**Comment by sedkis** [comment]
Duplicate of #2780

**Comment by maciejwojciechowski** [comment]
verified

**Comment by maciejwojciechowski** [comment]
what about case when key is created out of policy? I still see this 
INFO api: Can't retrieve key quota key=****2a36 message=key not found status=ok

**Comment by maciejwojciechowski** [comment]
verified key level
for policy level we have second issue reported. this one can be closed.

**Comment by sedkis** [comment]
This is a duplicate of #2780  and was resolved by #2789

**Comment by sedkis** [comment]
I cannot replicate on master as of Wed Feb 19th 2020, but I can recreate on v2.9.3
